webapp/src/main/java/com/box/l10n/mojito/smartling/SmartlingClient.java

```diff
@@ -0,0 +1,188 @@
+            nextRefresh = now.plus(Duration.ofMinutes(refreshDurationMinutes));
+            authToken = authenticationResponse.getResponse().getData().getAccessToken();
+            refreshToken = authenticationResponse.getResponse().getData().getRefreshToken();
+        }
+    }
+
+    AuthenticationResponse postForAuthentication(Map<String, Object> payload, String authApiV2) throws SmartlingClientException {
+        HttpEntity<Map<String, Object>> httpEntity = getHttpEntityForPayload(payload);
+        String postForObject = restTemplate.postForObject(
+                authApiV2, httpEntity, String.class);
+        AuthenticationResponse authenticationResponse;
+        try {
+            authenticationResponse = objectMapper.readValue(postForObject, AuthenticationResponse.class);
+        } catch (IOException e) {
+            logger.error(e.getMessage());
+            throw new SmartlingClientException(e.getMessage());
+        }
+        return authenticationResponse;
+    }
+
+    public void refresh() throws SmartlingClientException {
+        Instant now = Instant.now();
+        Instant nextCalculatedRefresh = now.plus(Duration.ofMinutes(refreshDurationMinutes));
+        if (nextRefresh.isAfter(now) && nextAuthentication.isAfter(now)) {
+            return;
+        } else if (nextCalculatedRefresh.isAfter(nextAuthentication)) {
+            authenticate();
+            return;
+        }
+        Map<String, Object> payload = getPayloadMapRefresh();
+        AuthenticationResponse authenticationResponse;
+        try {
+            authenticationResponse = postForAuthentication(payload, AUTH_API_V2_REFRESH);
+        } catch (SmartlingClientException e) {
+            throw new SmartlingClientException(e.getMessage());
+        }
+
+        if (authenticationResponse.isAuthenticationErrorResponse()) {
+            authenticate();
+        } else if (!authenticationResponse.isSuccessResponse()) {
+            nextRefresh = now;
+            throw new SmartlingClientException(authenticationResponse.getErrorMessage());
+        } else {
+            lastRefresh = now;
+            nextRefresh = nextCalculatedRefresh;
+            refreshToken = authenticationResponse.getResponse().getData().getRefreshToken();
+        }
+    }
+
+    public SourceStringsResponse getSourceStrings(String projectId, String file, Integer offset) throws SmartlingClientException {
+        refresh();
+        Map<String, Object> payload = getPayloadMapPullSourceString();
+
+        HttpEntity<Map<String, Object>> httpEntity = getHttpEntityForPayloadWithAuthorization(payload);
+
+        Map<String, String> pathVariables = new HashMap<>();
+        pathVariables.put("projectId", projectId);
+        pathVariables.put("fileUri", file);
+        pathVariables.put("offset", offset.toString());
+        UriTemplate template = new UriTemplate(API_PULL_SOURCE_STRINGS);
+        ResponseEntity<String> exchange = restTemplate.exchange(
```

Comment:
same as before, the restTemplate do the Jackson deserialization instead of getting a String and then converting with the object mapper.